lib/pka.c

```diff
@@ -844,7 +874,9 @@ void* polling_func(void* args)
            memset(&results, 0, sizeof(pka_results_t));
            init_results_operand(&results, 2, res1, MAX_BYTE_LEN, res2, MAX_BYTE_LEN);
 
-           pka_get_result((pka_handle_t) t_args->local_info, &results);
+	   // Passing USER_DATA_POLLING in order to handle locking. In async polling case here, lock is 
+	   // acquired already. pka_get_result_by_user_data call shall acquire lock only in sync context.
+           pka_get_result_by_user_data((pka_handle_t) t_args->local_info, &results, (void*)USER_DATA_POLLING);

```

Comment:
Why the user data MACRO is casted to a void pointer?

Reply:
user_data is void pointer in cmd and rslt structure.